lib/rubocop/cop/rspec/single_argument_message_chain.rb

```diff
@@ -33,11 +37,27 @@ def on_send(node)
         def autocorrect(node)
           lambda do |corrector|
             corrector.replace(node.loc.selector, replacement(node.method_name))
+            message_chain(node) do |arg|
+              autocorrect_hash_arg(corrector, arg) if single_key_hash?(arg)
+            end
           end
         end
 
         private
 
+        def autocorrect_hash_arg(corrector, arg)
+          key, value = *arg.children.first
+
+          corrector.replace(arg.loc.expression, key_to_arg(key))
+          corrector.insert_after(arg.parent.loc.end,
+                                 ".and_return(#{value.source})")
+        end
+
+        def key_to_arg(node)
+          key, = *node
+          ":#{key.to_sym}"
```

Comment:
At least, I don't think there's any case where `#to_sym` and then an implicit `#to_s` is different than just the implicit `#to_s`.

I've realized the code as it stands actually has a bug though.

Try adding:

```
include_examples(
  'autocorrect',
  'before { allow(foo).to receive_message_chain(:"#{foo}" => 42) }',
  'before { allow(foo).to receive(:"#{foo}").and_return(42) }'
)
```

to the test suite (you can also observe the same issue with an interpolated string rather than a symbol).

My suggestion fixes these issues, but I don't know if there are potentially any other corner cases.

We could also, of course, limit the node pattern to only match against sym, dsym, str, and dstr. The way i have it now seems to handle sends and constants though.

ex:

```
receive_message_chain(foo => 42)
```
receive_message_chain(FOO => 42)
```

I can't think of any other semi-reasonable case at the moment.